File: projetArmeeBack/src/main/java/projetArmee/projetArmeeBack/entities/TypeA.java
package projetArmee.projetArmeeBack.entities;

public enum TypeA {
//@JsonProperty("fusil_dassaut")
fusil_dassaut(10,10),
//@JsonProperty("mitraillette")
mitraillette(15,15),
//@JsonProperty("fusil_de_precision")
fusil_de_precision(5,5),
pistolet(5,5),pierrier(5,5),grenade(10,10),roquette(20,20),couteau(5,5),mine(10,10);
	
	private int prix;
	private int degat;

	private TypeA(int prix, int degat) {
		this.prix = prix;
		this.degat = degat;
	}

	public int getPrix() {
		return prix;
	}

	public void setPrix(int prix) {
		this.prix = prix;
	}

	public int getDegat() {
		return degat;
	}

	public void setDegat(int degat) {
		this.degat = degat;
	} 
}

File: projetArmeeAngular/projetArmee/src/app/models/type-a.ts
enum enumArme {
  fusil_dassaut = 'fusil_dassaut',
  mitraillette = 'mitraillette',
  fusil_de_precision = 'fusil_de_precision',
  pistolet = 'pistolet',
  pierrier = 'pierrier',
  grenade = 'grenade',
  roquette = 'roquette',
  couteau = 'couteau',
}

export interface TypeA {
  name: enumArme;
  prix: number;
  degat: number;
}

export const TypeAValues: TypeA[] = [
  { name: enumArme.fusil_dassaut, prix: 10, degat: 10 },
  { name: enumArme.mitraillette, prix: 15, degat: 15 },
  { name: enumArme.fusil_de_precision, prix: 5, degat: 5 },
  { name: enumArme.pistolet, prix: 5, degat: 5 },
  { name: enumArme.pierrier, prix: 5, degat: 5 },
  { name: enumArme.grenade, prix: 10, degat: 10 },
  { name: enumArme.roquette, prix: 20, degat: 20 },
  { name: enumArme.couteau, prix: 5, degat: 5 },
];
